sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -819,6 +829,33 @@ private async Task<AmqpSession> CreateAndOpenSessionAsync(AmqpConnection connect
             }
         }
 
+        private void ValidateCanCreateSenderLink(string entityPath)
+        {
+            if (_useSingleSession)
+            {
+                lock (_syncLock)
+                {
+                    // The send-via entity is a receiver and there are no active links. We are reconnecting the connection.
+                    if (_sendViaReceiverEntityPath != null && !ActiveLinks.Any())
```

Comment:
```suggestion
                    if (_sendViaReceiverEntityPath != null && ActiveLinks.Count == 0)
```

Using LINQ with a concurrent collection type forfeits thread-safety as a general rule.  There may be a special case for `Any()` but probably safer and more efficient to use the `Count` directly.

Reply:
Looks like IsEmpty is preferred:
  AmqpConnectionScope.cs(838, 63): [CA1836] Prefer 'IsEmpty' over 'Count' to determine whether the object contains or not any items